Source/Scene/ModelInstanceCollection.js

```diff
@@ -997,9 +998,15 @@ ModelInstanceCollection.prototype.update = function (frameState) {
     this._instancingSupported = context.instancedArrays;
     createModel(this, context);
     var that = this;
-    this._model.readyPromise.otherwise(function (error) {
-      that._state = LoadState.FAILED;
-      that._readyPromise.reject(error);
+    this._model.readyPromise.then(function (model) {
+      model.activeAnimations
+        .addAll({
+          loop: ModelAnimationLoop.REPEAT,
+        })
+        .otherwise(function (error) {
+          that._state = LoadState.FAILED;
+          that._readyPromise.reject(error);
+        });
```

Comment:
The `otherwise` should be chained after `.then` rather than after `.addAll`. This `.then`/`.otherwise` chain allows the promise to be handled correctly whether the promise is resolved or rejected.

Chaining it after `addAll` wouldn't end up catching an error since `addAll` does not return a promise. My mistake for missing this earlier, I think I got lost in the indendation.

```suggestion
    this._model.readyPromise
      .then(function (model) {
        model.activeAnimations.addAll({
          loop: ModelAnimationLoop.REPEAT,
        });
      })
      .otherwise(function (error) {
        that._state = LoadState.FAILED;
        that._readyPromise.reject(error);
      });
```

Reply:
Despite this code suggestion, I've been thinking more about the overall design here and think it makes sense to bring the functionality to `Instanced3DModel3DTileContent` instead. I didn't noticed until reviewing that `ModelInstanceCollection` already has an `activeAnimations` getter and it's the caller's responsibility whether to add animations or not.

So the change will be something like this: in `Instanced3DModel3DTileContent`, after `ModelInstanceCollection` is created, add the animations when the collection is ready. It'll look just like the `Batched3DModel3DTileContent` code in this PR except that model will be a model instance collection.